gremlin-go/driver/bytecode.go

```diff
@@ -19,7 +19,131 @@ under the License.
 
 package gremlingo
 
-type bytecode interface {
-	addSource(sourceName string, args ...interface{})
-	addStep(stepName string, args ...interface{})
+import "reflect"
+
+type bytecode struct {
+	sourceInstructions []instruction
+	stepInstructions   []instruction
+	bindings           map[string]interface{}
+}
+
+func newBytecode(bc *bytecode) *bytecode {
```

Comment:
I see why you would make a constructor, but what's the purpose of the bytecode struct being pass in? I see that you append the instructions of it onto the one you are creating... I am just wondering what the use case for that would be.

Reply:
GraphTraversalSource seems to use it in the Python implementation for returning an instanced copy of a traversal.